security/tasks.py

```diff
@@ -479,12 +480,17 @@ def apply(self, args=None, kwargs=None, related_objects=None, **options):
 
     def apply_async(self, args=None, kwargs=None, related_objects=None, **options):
         app = self._get_app()
-        if self.request.id or app.conf.task_always_eager:
-            return super().apply_async(args=args, kwargs=kwargs, related_objects=related_objects, **options)
-        else:
-            return self._first_apply(
-                is_async=True, args=args, kwargs=kwargs, related_objects=related_objects, **options
-            )
+        try:
+            if self.request.id or app.conf.task_always_eager:
+                return super().apply_async(args=args, kwargs=kwargs, related_objects=related_objects, **options)
+            else:
+                return self._first_apply(
+                    is_async=True, args=args, kwargs=kwargs, related_objects=related_objects, **options,
+                )
+        except (InterfaceError, OperationalError) as ex:
+            logger.error(str(ex))
```

Comment:
Maybe you could log the exception using `exception()` method? I am also thinking if this is really error, if you recover from it right away. Shouldn't you log just the INFO message? Or log nothing at all?

Reply:
I'd say this is at least a warning. Recovered or not, the database was unavailable and we should know about it.

Instead I've been thinking about whether we should try and rerun the task. @matllubos

Reply:
I think I agree with warning. Maybe you could put both textations in one warning log?